lib/rack/core_ext/hash.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Hash has `transform_keys` since Ruby 2.5.
+# For Ruby < 2.5, we need to add the following
+
+module Rack
+  module HashExtensions
+    if Gem::Version.new(RUBY_VERSION) < Gem::Version.new(2.5)
```

Comment:
Do we need to check the version given we are checking if `{}.respond_to?(:transform_keys)`?

Reply:
I was thinking of making it more explicit about what version of ruby will this be used for. But, I guess I already add it in the comment.

Reply:
Ok, removed.